app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaMetaDataExtension.java

```diff
@@ -40,11 +40,17 @@
 
     @Override
     public Optional<MetaData> meta(Map<String, Object> parameters) {
-        final String topic = ConnectorOptions.extractOption(parameters, "topic");
-        final String brokers = ConnectorOptions.extractOption(parameters, "brokers");
-        final String certificate = ConnectorOptions.extractOption(parameters, "brokerCertificate");
-        final String transportProtocol = ConnectorOptions.extractOption(parameters, "transportProtocol");
-        LOG.debug("Getting metadata from Kafka connection to {} with protocol {}", brokers, transportProtocol);
+        final String topic = ConnectorOptions.extractOption(parameters, KafkaBrokerService.TOPIC);
+        final String brokers = ConnectorOptions.extractOption(parameters, KafkaBrokerService.BROKERS);
+        final String certificate = ConnectorOptions.extractOption(parameters, KafkaBrokerService.BROKER_CERTIFICATE);
+        final String securityProtocol = ConnectorOptions.extractOption(parameters, KafkaBrokerService.SECURITY_PROTOCOL);
+        final String saslMechanism = ConnectorOptions.extractOption(parameters, KafkaBrokerService.SASL_MECHANISM);
+        final String saslLoginCallbackHandlerClass = ConnectorOptions.extractOption(parameters, KafkaBrokerService.SASL_LOGIN_CALLBACK_HANDLER_CLASS);
```

Comment:
Could we make it simpler on the user and pick one (say OAuth bearer) so they don't need figure out the right combination of parameters?

Reply:
I thought about having it as default (handler class and token uri) when user selects oauthbearer, but as I am not familiar with java kafka clients, I am not sure if those values could change a lot. I see kafka clients may have different sets of combinations for example scram, jwt, etc. But as you did notice this, I think it would be a good idea to have them as default and documenting it.